lib/kafka_ex/protocol/common.ex

```diff
@@ -0,0 +1,16 @@
+defmodule KafkaEx.Protocol.Common do
+  @moduledoc """
+  A collection of common request generation and response parsing functions for the
+  Kafka wire protocol.
+  """
+
+  @doc """
+  Generate the wire representation for a list of topics.
+  """
+  def topic_data([]), do: ""
+
+  def topic_data([topic|topics]) do
```

Comment:
Regardless of the above - let's not go overboard here, I'd say. You'll interact with a handful of topics at best, you'll never see a measurable difference even if TCO is a silver bullet. I'll eat my words and recat the function when a bug report gets filed, but without proof that someone needs to deal with a million topics, I think the cleaner code version should stay.